src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -95,7 +95,18 @@ private async Task AcquireTokenInternalAsync(IDictionary<string, string> brokerP
                 {
                     _logger.Verbose("User is specified for silent token request. Starting silent broker request");
                     string silentResult = await _brokerHelper.GetBrokerAuthTokenSilentlyAsync(brokerPayload, _activity).ConfigureAwait(false);
-                    s_androidBrokerTokenResponse = CreateMsalTokenResponseFromResult(silentResult);
+                    if (!string.IsNullOrEmpty(silentResult))
```

Comment:
This is not what I meant but it is a good catch, as returning null would probably create problems downstream.

What I meant is that `CreateMsalTokenResponseFromResult` only handles the happy path when the broker returns a successful result. If the result has an error, you need to extract it, like at line 170. Preferably without having to duplicate code.

Also, this method is now too big, please see if you can split it into smaller pieces.

Reply:
I see now looking upstream that you return NULL if the broker does not return a successfuil response. Would be better if we can extract the actual error like you do with in the interactive broker flow. If possible.